src/hotspot/share/prims/jvmtiGetLoadedClasses.cpp

```diff
@@ -118,7 +118,7 @@ JvmtiGetLoadedClasses::getClassLoaderClasses(JvmtiEnv *env, jobject initiatingLo
   LoadedClassesClosure closure(env, true);
   {
     // To get a consistent list of classes we need MultiArray_lock to ensure
-    // array classes aren't created during this walk. This walks through the
+    // array classes aren't created by another thread during this walk. This walks through the
```

Comment:
Thanks for clarifying, though I'm not sure why would care given the set of classes could have changed by the time we return them anyway.

Reply:
I think in this case, we might find an ObjArrayKlass without the mirror since the Klass is added to the higher_dimension links before the mirror is created.  The lock keeps them both together.